crates/chia-sdk-driver/src/layer.rs

```diff
@@ -32,13 +34,43 @@ pub trait Layer {
         ctx: &mut SpendContext,
         solution: Self::Solution,
     ) -> Result<NodePtr, DriverError>;
+
+    /// Creates a spend for this layer.
+    fn construct_spend(
+        &self,
+        ctx: &mut SpendContext,
+        solution: Self::Solution,
+    ) -> Result<Spend, DriverError> {
+        let solution = self.construct_solution(ctx, solution)?;
+        let puzzle = self.construct_puzzle(ctx)?;
+        Ok(Spend::new(puzzle, solution))
+    }
+
+    /// Creates a coin spend for this layer.
```

Comment:
I'd only make this available for outer layers

Reply:
![image](https://github.com/user-attachments/assets/4342267b-332a-4083-a0a8-a1b1bc4426da)